### PR TITLE
RX65N RTC Driver added

### PR DESCRIPTION
**Pull request by RB-tel**
RTC Driver files and related modifications added to the rx65n.

### CODE REVIEW

File: arch/renesas/src/rx65n/rx65n_rtc.h [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ***************************************************************************/
```
Comment (patacongo) [inline arch/renesas/src/rx65n/rx65n_rtc.h]: All new files added to the Apache repositories must have an Apache 2.0 license header with no copyright or author.  You can copy-paste the header from sched/sched/sched_getcpu.c.

File: boards/renesas/rx65n/rx65n-grrose/src/rx65n_appinit.c [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
```
Comment (patacongo) [inline boards/renesas/rx65n/rx65n-grrose/src/rx65n_appinit.c]: All new files added to the Apache repositories must have an Apache 2.0 license header with no copyright or author.  You can copy-paste the header from sched/sched/sched_getcpu.c.

File: boards/renesas/rx65n/rx65n-rsk2mb/src/rx65n_appinit.c [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
```
Comment (patacongo) [inline boards/renesas/rx65n/rx65n-rsk2mb/src/rx65n_appinit.c]: All new files added to the Apache repositories must have an Apache 2.0 license header with no copyright or author.  You can copy-paste the header from sched/sched/sched_getcpu.c.

**Comment by patacongo** [comment]
When you change the headers on all of the new files to the new Apache 2.0 license I will be happy to merge the code.  Sorry to trouble you with this, but these are the rules for new files since NuttX in an Apache project (We will eventually get all of the older, existing file headers changed too).

In general the code looks good.  Thanks!  It is, however, difficult to review large chanages online.  Did you run the code through tools/nxstyle to assure that it meets the coding standard?

**Comment by RB-tel** [comment]
Hello @patacongo,

Apache License Header Added to all the new files and the modified files. 

Please check and merge if OK.

**Comment by RB-tel** [comment]
> When you change the headers on all of the new files to the new Apache 2.0 license I will be happy to merge the code. Sorry to trouble you with this, but these are the rules for new files since NuttX in an Apache project (We will eventually get all of the older, existing file headers changed too).
> 
> In general the code looks good. Thanks! It is, however, difficult to review large chanages online. Did you run the code through tools/nxstyle to assure that it meets the coding standard?

Yes, we have checked for the coding style.

**Pull request merged**